src/transaction-types.ts

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-restricted-globals */
+
+export type Address = `0x${string}` | Buffer;
+
+export type Transaction = {
```

Comment:
It looks like this only includes the legacy transaction properties. We would want to accept newer style transactions too though, e.g. those with EIP1559 properties.

Perhaps we should have a type for each new EIP that adds transaction properties? There appear to be 3 referenced in the `@ethereumjs/tx` library. Then the type we use in the keyring interface can be all of them combined.

Reply:
should we do something like this?

```
type Transaction = (EIP1559Transaction | LegacyTransaction) & TransactionBase
type SignedTransaction = Transaction & TransactionSignature;

type TransactionSignature = {
    /**
   * EC recovery ID.
   */
  v: bigint | string | number | Buffer;

  /**
   * EC signature parameter.
   */
  r: bigint | string | number | Buffer;

  /**
   * EC signature parameter.
   */
  s: bigint | string | number | Buffer;

}

type EIP1559Transaction = {
    maxPriorityFeePerGas: bigint | string | number ;
    maxFeePerGas: bigint | string | number ;
    gasPrice?: never;
}

type LegacyTransaction = {
    gasPrice: bigint | string | number ;
    maxPriorityFeePerGas?: never;
    maxFeePerGas?: never ;
}

type TransactionBase = {
... rest of the tx fields
}
```

Reply:
That's a good approach we can take but it doesn't take in account EIP2930 and EIP4844 transactions. As you can see in the commit [6942609](https://github.com/MetaMask/utils/pull/74/commits/69426091961b24c343ecc4661176907ef2a8e3e0) each EIP builds on top of the previous one.

`Legacy Tx < EIP2930 Tx < EIP1559 Tx < EIP4844 Tx`

Reply:
I am also not sure of the best way to compose these types.

The current draft looks reasonable to me though, and for now it doesn't strongly matter (we won't be using these to differentiate between each type of transaction for now - we're just using them to establish a complete set of each possible property)

Reply:
`v` should be named `yParity`, since that's what's used in EIP-1559.

Reply:
[a590419](https://github.com/MetaMask/utils/pull/74/commits/a5904194472e1405136bc2ac1c7c0175ed736e59)